core/interop/source-generation/ComWrappersGeneration/Server/Exports.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Diagnostics;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using Microsoft.Win32;
+
+namespace Tutorial;
+
+public static unsafe class Exports
+{
+    /// <summary>
+    /// Returns a pointer to an IClassFactory instance that corresponds to the requested <paramref name="classId"/>.
+    /// <paramref name="interfaceId"/> is expected to be the IID of IClassFactory.
+    /// This method is called by the COM system when a client requests an object that this server has registered.
+    /// <see href="https://learn.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-dllgetclassobject"/>
+    /// </summary>
+    [UnmanagedCallersOnly(EntryPoint = nameof(DllGetClassObject))]
+    public static int DllGetClassObject(Guid* classId, Guid* interfaceId, nint* ppIClassFactory)
+    {
+        Console.WriteLine($"Server: Class ID requested from DllGetClassObject: {*classId}");
+        Console.WriteLine($"Server: Interface ID requested from DllGetClassObject: {*interfaceId}");
+        if (*classId != new Guid(ClsIds.Calculator)
+            || *interfaceId != new Guid(IClassFactory.IID))
+        {
+            *ppIClassFactory = 0;
+            return -2147221231; // 0x80040111 CLASS_E_CLASSNOTAVAILABLE
+        }
+        var factory = new ClassFactory();
```

Comment:
This should always be the same, right? Perhaps store it in a static local that can be reused.